Signin Page in vue app

AfaqAhmed-XGrid opened the pull request:
## **Issue** #18 
     Title: signin page in vue app

## Description
This pull request sets up the signin page i vue app:
=> Sign in page: Used quasar and custom scss to design sign in page that is fully responsive.
=> Http request: Used axios to make http request to sign in user.
=> Form validation: Added form validation using quasar
=> Protected Routes: Protected all routes except and sign in and sign up.

## **Checklist**
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the README.md, or the docs directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes (if applicable)
- [x] Every function, interface, class has a comment describing what it does and input/output parameters

AfaqAhmed-XGrid committed:
completed sign in page with all functionalities

This commit includes: styling, http requests, form validation, forgot password functionality and routes protection.

saif-xgrid commented on ui_vue/src/views/SignIn.vue:
```diff
@@ -15,6 +15,174 @@ limitations under the License.
 
 <template>
-  <h1>Sign In Page</h1>
-  <p>Comming soo</p>
```
Please fix this typo

AfaqAhmed-XGrid replied:
It is the part of the previous code. Now, it is changed to latest signin page code.

AfaqAhmed-XGrid committed:
added cypress testing adn resolved some error regarding forgot password functionality and callback page

resolved the issue of constants in callback vue page
improved ui of forgot password email and allocated separate variable for its email input

XGMqasim commented on ui_vue/cypress/e2e/signin.cy.ts:
```diff
@@ -0,0 +1,22 @@
+    cy.get('[data-cy="password"]').type('Abcd2067*');
+    cy.get('[data-cy="submit"]').click();
+    cy.url().should('include', '/sign-in');
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
```
Why disabled ESLint?

AfaqAhmed-XGrid replied:
I wanted to add some time between the two tests but eslint was giving error of unnecessary wait.
I added this time gap to observe proper ui changes during testing.

AfaqAhmed-XGrid committed:
added cypress testing adn resolved some error regarding forgot password functionality and callback page

resolved the issue of constants in callback vue page
improved ui of forgot password email and allocated separate variable for its email input

AfaqAhmed-XGrid committed:
Merge branch 'signin-page-in-vue-app' of https://github.com/AfaqAhmed-XGrid/Cine-Info-AfaqAhmed-XGrid into signin-page-in-vue-app

asad-xgrid commented on ui_vue/cypress.config.ts:
```diff
@@ -20,6 +20,7 @@ import { defineConfig } from 'cypress';
   e2e: {
     specPattern: 'cypress/e2e/**/*.{cy,spec}.{js,jsx,ts,tsx}',
-    baseUrl: 'http://localhost:4173'
+    baseUrl: 'http://localhost:5173',
```
You need to define the base url some where which is included in gitignore eg: .env

AfaqAhmed-XGrid replied:
Done

asad-xgrid commented on ui_vue/cypress/support/commands.ts:
```diff
@@ -1,39 +1,87 @@
+
+  const clientId = import.meta.env.VITE_API_GITHUB_CLIENT_ID;
+  const clientSecret = import.meta.env.VITE_API_GITHUB_CLIENT_SECRET;
+  const code = 'd4674f99a4b3dab1c053';
```
can you please what code used for and why is not placed in constants or env file

AfaqAhmed-XGrid replied:
this is temporary code. I have moved it to env and added comment

asad-xgrid commented on ui_vue/cypress/support/commands.ts:
```diff
@@ -1,39 +1,87 @@
+      Accept: 'application/json',
+    },
+  }).then(({ body }) => {
+    const accessToken = body.access_token;
```
what if the access token is not returned ?

AfaqAhmed-XGrid replied:
wrote an assertion there taht would check access_token key in body and if it is not there, test would be failed

asad-xgrid commented on ui_vue/src/composables/useNotify.ts:
```diff
@@ -0,0 +1,47 @@
+ * Function to toastify the message on success and failure scenarios while making http requests
+ * @return {Object} returns object of toast functions for success and error
+ */
+export default function useNotify(): {
```
Convert it to Arrow function

AfaqAhmed-XGrid replied:
Done

asad-xgrid commented on ui_vue/src/styles/colorMode.scss:
```diff
@@ -0,0 +1,19 @@
+}
+
+.dark-bg-color {
+    background-color: #3949AB;
```
move colors to the colors class

AfaqAhmed-XGrid replied:
Done

AfaqAhmed-XGrid committed:
added comments, changed vue app env variables to cypress env variables, improved folder structure for scss styling

Added comments in cypress files for better understanding the code

Added cypress dedicated env variables instead of env variables in whole vue app

moved light and dark mode styling to their respective files and deleted the dedicated file
